src/app/games/yinsh/yinsh.component.ts

```diff
@@ -322,13 +323,11 @@ export class YinshComponent
                 this.viewInfo.selectableCoords.push(coord);
             }
         }
-        switch (this.movePhase) {
-            case 'INITIAL_CAPTURE_SELECT_FIRST':
-                this.movePhase = 'INITIAL_CAPTURE_SELECT_LAST';
-                break;
-            case 'FINAL_CAPTURE_SELECT_FIRST':
-                this.movePhase = 'FINAL_CAPTURE_SELECT_LAST';
-                break;
+        if (this.movePhase === 'INITIAL_CAPTURE_SELECT_FIRST') {
+            this.movePhase = 'INITIAL_CAPTURE_SELECT_LAST';
+        } else {
+            assert(this.movePhase === 'FINAL_CAPTURE_SELECT_FIRST', 'moveToCaptureSelectLast did not expect to be called in movePhase' + this.movePhase);
```

Comment:
Utils.expectToBe?

Reply:
C'pas un default :/ ?

Reply:
hmm non mais c'est la valeur attendue, expected to be !

Reply:
Mais les expectToBe on les utilise que pour les default des switch, pas pour tout les assert dont la condition est un === si :D ?

Reply:
on pourrait ! Ici ce if/else c'est jamais qu'un switch en vrai non ? À voir ce qui est préférable, mais à la lecture, un `expectToBe` ça a du sens ici il me semble.

Reply:
ok, donc ça me semble acceptable que les if(x=a) else { assert(x==b) deviennent des expect tobe pour cause de "c't'un peu un switch" !